lib/datadog/tracing/distributed/baggage.rb

```diff
@@ -96,19 +103,36 @@ def encode_key(key)
         end
 
         def encode_value(value)
```

Comment:
Are you able to merge encode_key and encode_value into one method, with the parameter for SAFE_CHARACTERS_VALUE or SAFE_CHARACTERS_KEY?